### PR TITLE
feat(ios, android): add tabBarVisible API

### PR DESCRIPTION
**Pull request by hansemannn**
Example:

```js
const isAndroid = Ti.Platform.osname === 'android';
let tabGroup;
let isVisible = true;

const window = Ti.UI.createWindow({
	title: 'Window 1',
	backgroundColor: '#fff'
});

const btn = Ti.UI.createButton({
	title: 'Show / hide tab group'
});

btn.addEventListener('click', () => {
	isVisible = !isVisible;
	tabGroup.tabBarVisible = isVisible;
});

window.add(btn);

tabGroup = Ti.UI.createTabGroup({
	style: isAndroid ? Ti.UI.Android.TABS_STYLE_BOTTOM_NAVIGATION : undefined,
	tabs: [
		Ti.UI.createTab({ window, title: window.title })
	]
});

tabGroup.open();
```

### CODE REVIEW

File: apidoc/Titanium/UI/TabGroup.yml [hunk 1]
```diff
@@ -588,6 +588,12 @@ properties:
     platforms: [iphone, ipad, macos]
     since: { iphone: "3.1.0", ipad: "3.1.0" }
 
+  - name: tabBarVisible
+    summary: Programmatically shows / hides the bottom tab bar of the tab group.
+    type: Boolean
+    platforms: [iphone, ipad, macos]
+    since: "11.1.0"
```
Comment (m1ga) [inline apidoc/Titanium/UI/TabGroup.yml:595]:
```suggestion
    since: "12.1.0"
```

File: apidoc/Titanium/UI/TabGroup.yml [hunk 2]
```diff
@@ -588,6 +588,12 @@ properties:
     platforms: [iphone, ipad, macos]
     since: { iphone: "3.1.0", ipad: "3.1.0" }
 
+  - name: tabBarVisible
+    summary: Programmatically shows / hides the bottom tab bar of the tab group.
+    type: Boolean
+    platforms: [iphone, ipad, macos]
```
Comment (m1ga) [inline apidoc/Titanium/UI/TabGroup.yml:594]:
```suggestion
    platforms: [android, iphone, ipad, macos]
```

**Comment by m1ga** [comment]
Android has `disableTabNavigation` (https://titaniumsdk.com/api/titanium/ui/tabgroup.html#methods_disabletabnavigation) already and I've added animation support in https://github.com/tidev/titanium_mobile/pull/11681
Think we can close mine if this one gets merged.

**Comment by hansemannn** [comment]
All updated!

**Comment by m1ga** [comment]
is the behavior intended:

the current `disableTabNavigation`:

https://user-images.githubusercontent.com/4334997/224561630-e8cec7e7-86d2-4a54-b410-cac82780ba04.mp4

this PR:

https://user-images.githubusercontent.com/4334997/224561557-0118c4ae-c019-4582-91aa-7455328cda67.mp4

the window is not extending

**Comment by m1ga** [comment]
@hansemannn 
you might want to look at the changes in https://github.com/tidev/titanium_mobile/pull/11681/files#diff-88df67787e4e1b86cf521062eeb89d4e91082c17f403cc6623f36388e0dc7d7c
and add those too to extend the window.
Or map your new property to use that function (with the changes of my PR to animate it)

**Comment by hansemannn** [comment]
@m1ga So only `this.insetsProvider.setBottomBasedOn(this.mBottomNavigationView);` is missing? Would need some more time to check that

**Comment by m1ga** [comment]
I'm sure this part too https://github.com/tidev/titanium_mobile/blob/ba3cda4469ce082a63497f4beb4830b2dee995f8/android/modules/ui/src/java/ti/modules/titanium/ui/widget/tabgroup/TiUIBottomNavigationTabGroup.java#L204-L220
but I would need to look at that too.

And also test https://github.com/tidev/titanium_mobile/pull/11681#issuecomment-623785486 (+ following replies in that post).

I could also update my PR to be used with `tabBarVisible` too. Otherwise it would be nicer to make parity for `disableTabNavigation` since that is available for Android for a while

**Comment by hansemannn** [comment]
Addressed in https://github.com/tidev/titanium_mobile/commit/2d5d7f8bc3a4f9ad22828c7bf56a2bf81dafa392

**Comment by hansemannn** [comment]
For maximum coverage, we could add `animated` as a property and make it cross-platform as well.

**Comment by m1ga** [comment]
https://github.com/tidev/titanium_mobile/pull/13770
to extend the old method and internally use your method. I'll close my old PR now since that is basically the same now :smile:

**Pull request merged**